App settings support

dgarciarubio opened the pull request:
I was required to configure these enrichers from app settings for a project, so I decided to add a PR to share my work.

Related issue [#34](https://github.com/serilog-web/classic/issues/34)

dgarciarubio committed:
Fix broken xml comment

dgarciarubio committed:
Add configuration extensions for ClaimsValueEnricher

dgarciarubio committed:
Add configuration extensions for HttpRequestClientHostIPEnricher

dgarciarubio committed:
Add configuration extensions for HttpRequestClientHostNameEnricher

dgarciarubio committed:
Add configuration extensions for HttpRequestIdEnricher

dgarciarubio committed:
Add configuration extensions for HttpRequestNumberEnricher

dgarciarubio committed:
Add configuration extensions for HttpRequestRawUrlEnricher

dgarciarubio committed:
Add configuration extensions for HttpRequestTraceIdEnricher

dgarciarubio committed:
Add configuration extensions for HttpRequestTypeEnricher

dgarciarubio committed:
Add configuration extension support for HttpRequestRawUrlEnricher and HttpRequestUrlEnricher within the same method.

dgarciarubio committed:
Add configuration extensions for HttpRequestUrlReferrerEnricher

dgarciarubio committed:
Add configuration extensions for HttpRequestUserAgentEnricher

dgarciarubio committed:
Add configuration extensions for HttpSessionIdEnricher

dgarciarubio committed:
Add confioguration extensions for UserNameEnricher

dgarciarubio committed:
Remove unnecesary method overloads for consistency.

dgarciarubio committed:
Updated web project to use app settings

dgarciarubio committed:
Separated support for Raw Url and Url into different extension methods, allowing both of them to be configurated simultaneously in the app settings file

dgarciarubio committed:
Updated web project to use asp net classic enrichers in app settings

tsimbalar commented on src/SerilogWeb.Classic/SerilogWebClassicLoggerConfigurationExtensions.cs:
```diff
@@ -0,0 +1,186 @@
+            this LoggerEnrichmentConfiguration enrichmentConfiguration)
+        {
+            if (enrichmentConfiguration == null) throw new ArgumentNullException(nameof(enrichmentConfiguration));       
+            return enrichmentConfiguration.With< HttpRequestUrlEnricher>();
```
nitpicking : there is a whitespace between `With<` and `HttpRequestUrlEnricher` :)

tsimbalar commented:
Hi Daniel,

Thanks for your PR, it looks pretty good to me !

If you have time, it would also be good to update the README file (in the [*Enrichers*](https://github.com/serilog-web/classic#enrichers) section : 
- use `WithXXX()` instead of the the *enricher's* class name in the list of provided *Enrichers* 
- use the `WithXXX()` method in the code sample
- to at least mention that the enrichers can be defined also in config file with *Serilog.Settings.AppSettings* and put an example (taking the `appSettings` section from ` test/SerilogWeb.Test/Web.config` is probably good enough)

When this is done, I'll increment the version number and publish a new version.

Thanks a lot !

dgarciarubio commented:
I will probably be able to do so this week. I was also not sure wether the extensions class should go inside the Classic folder or not... I guess it is fine outside...?

tsimbalar commented:
Yes, that one is a bit special, we do want it in the `Serilog` namespace as you did it, and our root namespace is `SerilogWeb`, so there's no ideal place. 

I think your choice is just fine, and the `Classic` folder is already pretty crowded ;)

dgarciarubio committed:
Update README.md to reflect 'Enrich.WithXXX' APIs

Updated the Enrichers section and the code examples to reflect the new APIs.
Added sample config file configuration.

tsimbalar commented:
Awesome ! 

Here is the new version on Nuget : https://www.nuget.org/packages/SerilogWeb.Classic/4.2.41 

Congratulations on your first merged PR ! 🎉

tsimbalar commented:
If you feel like adding the same kind of improvements in *SerilogWeb.Classic.Mvc* and *SerilogWeb.Classic.WebApi* , help is more than welcome ;) 
(see serilog-web/classic-mvc#8 and serilog-web/classic-webapi#14 )
